dns/dns_updater.go

```diff
@@ -226,7 +230,11 @@ func (u *updater) createChanges(hostToIngress hostToIngress,
 			continue
 		}
 
-		changes = append(changes, newChange("UPSERT", host, dnsDetails.dnsName, dnsDetails.hostedZoneID))
+		matchingRecordSet, recordExists := hostToRecords[host]
+		elbHasChanged := recordExists && dnsDetails.dnsName != *matchingRecordSet.AliasTarget.DNSName
```

Comment:
If you make hostToRecord have the key `{host, elbDNSName}` you don't need any of this logic.